src/main/java/com/zaxxer/hikari/pool/PoolElf.java

```diff
@@ -183,9 +183,10 @@ void setupConnection(final Connection connection, final long connectionTimeout)
       connection.setAutoCommit(isAutoCommit);
       connection.setReadOnly(isReadOnly);
 
-      final int defaultLevel = connection.getTransactionIsolation();
-      transactionIsolation = (transactionIsolation < 0 ? defaultLevel : transactionIsolation);
-      if (transactionIsolation != defaultLevel) {
+      if (transactionIsolation == -1) {
+         transactionIsolation = connection.getTransactionIsolation();
+      }
+      else if (transactionIsolation != Connection.TRANSACTION_NONE) {
```

Comment:
>
> http://docs.oracle.com/javase/7/docs/api/java/sql/Connection.html#setTransactionIsolation(int)
> specially notes not to call method with TRANSACTION_NONE


That convinces me further that the original code was more correct.  The
fact that a driver *can* return TRANSACTION_NONE from
*getTransactionIsolation()* is an indication by that driver that
transactions are not supported.

We therefore *must* *always* call *getTransactionIsolation()*, and even if
the user has specified READ_COMMITTED or some other isolation level, that
must be ignored and *transactionIsolation* should be set to
TRANSACTION_NONE.

This existing code is correct, except that this line should read:

transactionIsolation = (transactionIsolation < 0 || defaultLevel ==
Connection.TRANSACTION_NONE) ? defaultLevel : transactionIsolation);